homeassistant/components/lifx/coordinator.py

```diff
@@ -156,3 +202,43 @@ async def async_set_color_zones(
                 apply=apply,
             )
         )
+
+    async def async_update_wifi_info(self) -> None:
+        """Get the bulb's wifi signal strength."""
+        _LOGGER.debug("Updating wifi signal strength for %s", self.device.label)
+        try:
+            wifi_info = await async_execute_lifx(self.device.get_wifiinfo)
+            self._rssi = int(math.floor(10 * math.log10(wifi_info.signal) + 0.5))
+        except asyncio.TimeoutError as ex:
+            raise UpdateFailed(
+                f"Failed to fetch RSSI sensor data from device: {self.device.ip_addr}"
+            ) from ex
+
```

Comment:
We shouldn't poll this unless at least one entity is enabled that needs it

Reply:
Yeah, I wasn't sure how to work that out from the coordinator. All my attempts to override `async_update` failed.

Reply:
You could keep track of it by adding having the entity tell the coordinator it wants it in `async_added_to_hass` and tell it, it no longer wants it in `async_will_remove_from_hass`

Similar example https://github.com/home-assistant/core/blob/6da25c733e014624564140a761af5201e3bf0adc/homeassistant/components/unifiprotect/entity.py#L217